configuration.go

```diff
@@ -46,6 +46,9 @@ type GlobalConfiguration struct {
 	Boltdb                    *provider.BoltDb        `description:"Enable Boltdb backend"`
 	Kubernetes                *provider.Kubernetes    `description:"Enable Kubernetes backend"`
 	Mesos                     *provider.Mesos         `description:"Enable Mesos backend"`
+	CustomBackendHTTPError    bool                    `description:"Use custom HTTP response on backend service 404"`
+	CustomBackendErrorMessage string                  `description:"Custom error message to send in HTTP response if backend service returns a 404."`
+	CustomBackendErrorCode    int                     `description:"Custom HTTP status code to return if a backend service returns 404."`
```

Comment:
I would add `CustomBackendErrorMessage` and `CustomBackendErrorCode` in a struct `ErrorCodes`. This would avoid the need of `CustomBackendHTTPError`.
```
...
	Boltdb                    *provider.BoltDb        `description:"Enable Boltdb backend"`
	Kubernetes                *provider.Kubernetes    `description:"Enable Kubernetes backend"`
	Mesos                     *provider.Mesos         `description:"Enable Mesos backend"`
	ErrorCodes                *ErrorCodes             `description:"Custom error codes"`
}
```
WDYT?

Reply:
@emilevauge I'm thinking about reviving this PR...
Would you clarify if the following structure of `ErrorCodes` will be good for the PR:

```
type ErrorCodes struct {
  BackendErrorCode int `description:"HTTP error code returned when necessary backend is not found (404 by default)"`
  BackendErrorMessage string `description:"HTTP status line message when necessary backend is not found"`
}
```